dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java

```diff
@@ -390,37 +388,52 @@ public String extractFile(InputStream inputStream, String fileType) {
 
             Files.copy(inputStream, tempFile, StandardCopyOption.REPLACE_EXISTING);
 
-            zipFileSystem = FileSystems.newFileSystem(tempFile, (ClassLoader) null);
-            Path root = zipFileSystem.getPath("/");
-            Files.walk(root)
-                    .forEach(path -> {
-                        try {
-                            long fileSize = Files.size(path);
-                            if (Files.isDirectory(path)) {
-                                filePaths.add(path.toString().substring(1) + "/|" + fileSize );
-                            } else {
-                                filePaths.add(path.toString().substring(1) + "|" + fileSize );
-                            }
-                        } catch (IOException e) {
-                            e.printStackTrace();
+            if ("tar".equals(fileType)) {
```

Comment:
can't we use less duplicated code? try to copy and paste to github copilot or chatgpt

Reply:
there are many ways how to do it 
- unify parts to one generic function 
- use classes
- ...